rust/nasl-builtin-knowledge-base/src/lib.rs

```diff
@@ -4,26 +4,32 @@
 
 use std::time::{SystemTime, UNIX_EPOCH};
 
-use nasl_builtin_utils::{error::FunctionErrorKind, get_named_parameter, NaslFunction};
+use nasl_builtin_utils::{error::FunctionErrorKind, NaslFunction};
 use storage::{Field, Kb, Retrieve};
 
 use nasl_builtin_utils::{Context, Register};
+use nasl_function_proc_macro::nasl_function;
 use nasl_syntax::NaslValue;
 
-/// NASL function to set a knowledge base
-fn set_kb_item(register: &Register, c: &Context) -> Result<NaslValue, FunctionErrorKind> {
-    let name = get_named_parameter(register, "name", true)?;
-    let value = get_named_parameter(register, "value", true)?;
-    let expires = match get_named_parameter(register, "expires", false) {
-        Ok(NaslValue::Number(x)) => Some(*x),
-        Ok(NaslValue::Exit(0)) => None,
-        Ok(x) => {
+/// NASL function to set a value under item in a knowledge base
+/// If the already exists, it will be remove first to avoid duplications.
```

Comment:
```suggestion
/// NASL function to set a value under name in a knowledge base.
/// If the name already exists, the value is added to its list.
```
Not quite sure about that, but I guess we just push the value to the existing entry.